pkg/git/git.go

```diff
@@ -0,0 +1,222 @@
+package git
```

Comment:
At first glance this package looks like a copy of `github.com/weaveworks/eksctl/pkg/git`. In the recent weeks we've been working hard to reduce code duplication in favor of git module reuse, so it would be better to reuse the package `git` as a Go module instead of copying it over to `wksctl`.

Reply:
I'd like to. But during the development of this PR, I found that I needed the different behaviour from `github.com/weaveworks/eksctl/pkg/git`. That's why it's copied this way.

The different behaviours are (`eksctl` vs `wksctl`):
- `Rm` vs `RmRecursive`
- `Commit` vs `Commit` with default user and email.
- `Clone` specific branch vs `Clone` specific branch or tag or revision.

Reply:
I agree to do so in the long term.

Reply:
I was thinking how to get rid of the duplication and eventually came up with two thoughts:

1. I realized that `profile enable` is very similar to a native Git feature `git subtree`:

    ```bash
    git subtree add \
      --squash \
      --prefix profiles/github.com/chanwit/podinfo-profile \
      https://github.com/chanwit/podinfo-profile \
      v0.1.0
    
    git push # optionally
    ```    
    which would save us from having to clone any repository whatsoever.

2. The code in this PR manipulates the Git worktree index, creates a commit and optionally pushes the desired changes plus (possibly non-empty) preexisting changes that are in the index. This feels dangerous to me - if `wksctl` is to create commits on behalf of the user, we should prevent them from shooting themselves in the foot like this.

Given the above, I have the following suggestion for a way to address this comment:
- make a slight change in behavior: always make a commit in the local repo (ensuring that there are no staged changes) and replace the `--[no-]commit` flag with a `--[no-]push` flag which would control whether `wksctl` pushes the config or not.
- modify `profile enable` to perform `git subtree add` as above (and `git push` if chosen by the user)
- modify `profile disable` to perform `git rm -r -- <directory>` `git commit -m "disable profile <name>"` (and `git push` if chosen by the user)

This would give us:
- no need to modify or even use `github.com/weaveworks/eksctl/git` (because all git operations would be essentially one-liners plus a small function similar to `git.exec`) + no compatibility issues
- better tracking of changes (because changes performed by wksctl would be visible in history)
- prevention of accidentally pushing of files staged but not committed yet

Reply:
This is awesome. I'll try `git subtree` and get back to you!

Reply:
changed to `git subtree` and it works really well. thank you.